docs/developer-docs/security/security-best-practices/iam.mdx

```diff
@@ -105,3 +105,61 @@ The [auth-client](https://github.com/dfinity/agent-js/tree/main/packages/auth-cl
+
+This risk can be addressed by adopting the following remediations shown in the sequence diagram and explained further below.
+
+* Introduce an intermediate session key which is generated and stored by the web app proxy front-end.
+* Initiate the II client authentication protocol using this intermediate session key. By using a new session key which the attacker can’t control, the delegation issued by II would no longer be usable by the attacker if it were stolen in step 8, as the attacker doesn’t have access to the intermediate session private key.
+* [Create a delegation chain](/docs/current/references/ic-interface-spec#authentication) to allow the mobile application to use their session key. The delegation chain consists of two delegations as shown in the figure below. The first one delegates from the II canister key to the intermediate key and is generated by the II canister. The second one delegates from the intermediate key to the mobile app public key and is signed by the proxy front-end’s intermediate session private key. Note, this means the intermediate key can impersonate the user. Since the proxy front-end is served from the IC, it can be trusted to handle this key properly. It is up to the developer to ensure the confidentiality of this key. For example, using the WebCrypto API to create unextractable keys as is used internally by agent-js. Ideally this intermediate key is short lived to reduce the risk of exposure.
+* Return the delegation chain to the mobile app using [app links](https://developer.android.com/training/app-links) on Android and [universal links](https://developer.apple.com/documentation/xcode/allowing-apps-and-websites-to-link-to-your-content) on iOS. These mechanisms bind the domain name/hostname to the mobile app which prevents an attacker from using a malicious mobile app to receive the delegation chain. The domain to mobile app binding occurs through a JSON file that has to be hosted under the /.well-known directory of your web application. See [iOS](https://developer.apple.com/documentation/xcode/supporting-associated-domains)
+and [Android](https://developer.android.com/training/app-links/verify-android-applinks) documentation for details.
+* Return the delegation chain to the mobile app using a [URI fragment](https://www.w3.org/DesignIssues/Fragment.html) (everything following the # in the URI). The browser will load the URI if the mobile app linked to the app/univeral link isn't installed on the mobile device. The benefit of URI fragments is that they are not included in the request to the server if the browser were to resolve the URI. A URL parameter or path would be included in such a request and therefore it would leak the delegation chain to the proxy app back-end (most likely the IC boundary and replica nodes). A URI fragment is still available to the mobile app for extraction.
```

Comment:
```suggestion
* Return the delegation chain to the mobile app using a [URI fragment](https://www.w3.org/DesignIssues/Fragment.html) (everything following the # in the URI). The browser will load the URI if the mobile app linked to the app/univeral link isn't installed on the mobile device. The benefit of URI fragments is that they are not included in the request to the server if the browser were to resolve the URI. A URL parameter or path would be included in such a request and therefore it would leak the delegation chain to the proxy app backend (most likely the IC boundary and replica nodes). A URI fragment is still available to the mobile app for extraction.
```